src/workerd/api/tests/js-rpc-test.js

```diff
@@ -483,3 +483,17 @@ export let crossContextSharingDoesntWork = {
     await assert.rejects(() => env.MyService.tryUseGlobalRpcPromisePipeline(), expectedError);
   },
 }
+
+// Test that exceptions thrown from async native functions have a proper stack trace. (This is
+// not specific to RPC but RPC is a convenient place to test it since we can easily define the
+// callee to throw an exception.)
+export let testAsyncStackTrace = {
+  async test(controller, env, ctx) {
+    try {
+      await env.MyService.throwingMethod();
```

Comment:
Non-blocking: Not sure if this is tested elsewhere, but a test verifying the behavior when the remote throws an arbitrary JS value that is not an error object, and making sure it comes through here correctly would be also be good.

Reply:
I believe non-error values get stringified and wrapped in `Error`. Of course, this behavior long predates RPC, and is also not really relevant to this particular PR.